From f8b7abe36203ca60b525ae1cde5bb02287e2f3cc Mon Sep 17 00:00:00 2001
From: Patrick Williams <patrick@stwcx.xyz>
Date: Mon, 24 Feb 2025 17:10:48 -0500
Subject: [PATCH 330/330] hwmon: ina2xx: switch regmap cache to flat

We are getting kernel panics in the ina2xx driver.  There are a large
number of fixes since 6.6 in the REGMAP_MAPLE.  Let's asssume it
wasn't stable in 6.6 and instead use the REGMAP_FLAT.

Signed-off-by: Patrick Williams <patrick@stwcx.xyz>
---
 drivers/hwmon/ina2xx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwmon/ina2xx.c b/drivers/hwmon/ina2xx.c
index 345fe7db9de9..7d03fc0e82d3 100644
--- a/drivers/hwmon/ina2xx.c
+++ b/drivers/hwmon/ina2xx.c
@@ -130,7 +130,7 @@ static const struct regmap_config ina2xx_regmap_config = {
 	.use_single_write = true,
 	.use_single_read = true,
 	.max_register = INA2XX_MAX_REGISTERS,
-	.cache_type = REGCACHE_MAPLE,
+	.cache_type = REGCACHE_FLAT,
 	.volatile_reg = ina2xx_volatile_reg,
 	.writeable_reg = ina2xx_writeable_reg,
 };
-- 
2.47.1

